datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapCache.java

```diff
@@ -133,15 +132,14 @@ private int validateAndGetCacheSize() {
    */
   private List<BloomDMModel> loadBloomDataMapModel(CacheKey cacheKey) {
     DataInputStream dataInStream = null;
-    ObjectInputStream objectInStream = null;
     List<BloomDMModel> bloomDMModels = new ArrayList<BloomDMModel>();
     try {
       String indexFile = getIndexFileFromCacheKey(cacheKey);
       dataInStream = FileFactory.getDataInputStream(indexFile, FileFactory.getFileType(indexFile));
-      objectInStream = new ObjectInputStream(dataInStream);
       try {
-        BloomDMModel model = null;
-        while ((model = (BloomDMModel) objectInStream.readObject()) != null) {
+        while (dataInStream.available() > 0) {
```

Comment:
I've checked the `available` method when I contributed these lines of code and found it's not suitable here.
Better to keep them as it is.
Maybe you can check it again.

Reply:
I don't know the problem you are facing, but the old cannot work as we are not doing object serialization here.

Reply:
The javadoc said
```
Returns an estimate of the number of bytes that can be read (or skipped over)
from this input stream without blocking by the next invocation of a method
for this input stream. 
```
So `dataInStream.available()== 0` does not mean that we have already read all the content from the stream.

Reply:
It seems that you use `dataInStream.available()== 0` to indicate that all the content of the stream has been consumed, however it only means that you have only consumed all the available content with non-blocking, there may be leftover content in the stream which will be reached for the next blocking call.

Reply:
That's my understanding of the javadoc.
If the datamap file is small, it may not cause problem, but I insist that it will cause problem in some scenario.